book_source/developers_guide/Testing.Rmd

```diff
@@ -0,0 +1,330 @@
+PEcAn uses the testthat package developed by Hadley Wickham. Hadley has
+written instructions for using this pakcage in his
+[Testing](http://adv-r.had.co.nz/Testing.html) chapter.
+
+Rationale
+---------
+
+*  makes development easier
+*  provides working documentation of expected functionality
+*  saves time by allowing computer to take over error checking once a
+    test has been made
+*  improves code quality
+*  Further reading: [Aruliah et al 2012 Best Practices for Scientific
+    Computing](http://arxiv.org/pdf/1210.0530v3.pdf)
+
+Tests makes development easier and less error prone
+---------------------------------------------------
+
+Testing makes it easier to develop by organizing everything you are
+already doing anyway - but integrating it into the testing and
+documentation. With a codebase like PEcAn, it is often difficult to get
+started. You have to figure out
+
+*  what was I doing yesterday?
+*  what do I want to do today?
+*  what existing functions do I need to edit?
+*  what are the arguements to these functions (and what are examples of
+    valid arguments)
+*  what packages are affected
+*  where is a logical place to put files used in testing
+
+Quick Start:
+------------
+
+* decide what you want to do today
+* identify the issue in redmine or github (if none exists, create one)
+* to work on issue 99, create a new branch called “redmine99”,
+“github99” or some descriptive name… Today we will add enable an
+existing function, `make.cheas` to make `goat.cheddar`. We will know
+that we are done by the color and taste.
+
+    ```bash
+    git branch goat-cheddar
+    git checkout goat-cheddar
+    ```
+    * open existing (or create new) file in `inst/tests/`. If working on code in "myfunction" or a set of functions in "R/myfile.R", the file should be named accordingly, e.g. "inst/tests/test.myfile.R"
+    * if you are lucky, the function has already been tested and has some examples.
+    * if not, you may need to create a minimal example, often requiring a settings file. The default settings file can be obtained in this way:
+      ```r
+      settings <- read.settings(system.file("extdata/test.settings.xml", package = "PEcAn.utils"))
+      ```
+    * write what you want to do
+      ```r
```

Comment:
This is a pretty silly example and also a hard one to follow. We should replace with with a legitimate example (e.g. from our own code) or, better yet, point to a guide for writing tests, like the one written by Hadley.